csharp/src/Ice/Proxy.cs

```diff
@@ -277,7 +277,7 @@ public static Connection GetConnection(this IObjectPrx prx)
             {
                 response = await proxy.InvokeAsync(forwardedRequest, oneway, progress).ConfigureAwait(false);
             }
-            catch (DispatchException ex)
+            catch (RemoteException ex)
```

Comment:
Is this still needed after your retry refactoring?
What we really want here is to not unmarshal / throw remote exceptions InvokeAsync, and just keep them in marshaled form inside response.

Reply:
I will update that in a separate PR